src/main/scala/org/broadinstitute/dsde/firecloud/service/EntityService.scala

```diff
@@ -0,0 +1,78 @@
+package org.broadinstitute.dsde.firecloud.service
+
+/**
+ * Created by mbemis on 7/10/15.
+ */
+
+import javax.ws.rs.Path
+
+import akka.actor.{Actor, Props}
+import com.wordnik.swagger.annotations._
+import org.broadinstitute.dsde.firecloud.{EntityClient}
+import org.broadinstitute.dsde.firecloud.model.{WorkspaceEntity, WorkspaceIngest}
+import org.broadinstitute.dsde.firecloud.model.ModelJsonProtocol._
+import org.slf4j.LoggerFactory
+import spray.http.StatusCodes._
+import spray.httpx.SprayJsonSupport._
+import spray.routing._
+import org.broadinstitute.dsde.vault.common.directives.OpenAMDirectives._
+
+class EntityServiceActor extends Actor with EntityService {
+  def actorRefFactory = context
+  def receive = runRoute(routes)
+}
+
+@Api(value = "/workspaces/{workspaceNamespace}/{workspaceName}/entities", description = "Entity Service", position = 2, produces = "application/json")
+trait EntityService extends HttpService with FireCloudDirectives {
+
+  private final val ApiPrefix = "entities"
+  private implicit val executionContext = actorRefFactory.dispatcher
+
+  val routes =  optionsRoute ~
+                listEntitiesPerTypeRoute
+
+  lazy val log = LoggerFactory.getLogger(getClass)
+
+  @ApiOperation(
+    value = "entities options",
+    nickname = "entitiesOptions",
+    httpMethod = "OPTIONS",
+    notes = "response is an OPTIONS response")
+  @ApiResponses(Array(
+    new ApiResponse(code = 200, message = "OK")))
+  def optionsRoute: Route =
+    path(ApiPrefix) {
```

Comment:
Never mind ... I see what this is doing - it's only for the options request. In that case - it is confusing to me to have the top-level /entities path handled at all. Since an entity is only nested inside a workspace, it makes more sense to me to have this use the same path as handled below (path("workspaces" / Segment / Segment / "entities")) which would work for any provided namespace/name segment provided, or even shorten that to "workspaces/entities".